### PR TITLE
KT-14095 Eclipse: Outline view could show implicit types too

### PR DESCRIPTION
**Pull request by SimonScholz**
Solved by obtaining a org.jetbrains.kotlin.descriptors.CallableDescriptor from a org.jetbrains.kotlin.resolve.BindingContext

Change-Id: I158d7d44b8f1724f7fbffd4effcd1873747e2404
Signed-off-by: Simon Scholz <simon.scholz@vogella.com>

### CODE REVIEW

**Comment by SimonScholz** [comment]
image

Thanks for the hint with the DescriptorRenderer. Looks like expected

File: kotlin-eclipse-ui/src/org/jetbrains/kotlin/ui/editors/outline/PsiLabelProvider.java [hunk 1]
```diff
@@ -103,6 +113,8 @@ private String getPresentableElement(KtElement declaration) {
                     KtTypeReference typeReference = parameter.getTypeReference();
                     if (typeReference != null) {
                         text += typeReference.getText();
+                    } else {
+                        text += computeReturnType(function);
```
Comment (zarechenskiy) [inline kotlin-eclipse-ui/src/org/jetbrains/kotlin/ui/editors/outline/PsiLabelProvider.java]: Don't hurry, this type reference belongs to parameter, not return type :) 
You could check function like this:
`fun foo() = 123`, there is no return type in the Outline view

File: kotlin-eclipse-ui/src/org/jetbrains/kotlin/ui/editors/outline/PsiLabelProvider.java [hunk 2]
```diff
@@ -120,4 +132,19 @@ private String getPresentableElement(KtElement declaration) {
         
         return text;
     }
+    
+    private String computeReturnType(KtDeclaration ktDeclaration) {
+        KtFile ktFile = ktDeclaration.getContainingKtFile();
+        BindingContext bindingContext = KotlinAnalysisFileCache.INSTANCE.getAnalysisResult(ktFile).getAnalysisResult().getBindingContext();
+        DeclarationDescriptor declarationDescriptor = bindingContext.get(BindingContext.DECLARATION_TO_DESCRIPTOR, ktDeclaration);
+        if(declarationDescriptor instanceof CallableDescriptor) {
```
Comment (zarechenskiy) [inline kotlin-eclipse-ui/src/org/jetbrains/kotlin/ui/editors/outline/PsiLabelProvider.java]: Minor: formatting, there should be space after `if` keyword

**Comment by zarechenskiy** [comment]
Merged manually. Thank you!

**Pull request closed without merging**